test/images/agnhost/resource-consumer-controller/controller.go

```diff
@@ -43,8 +45,28 @@ var (
 	consumerPort             int
 	consumerServiceName      string
 	consumerServiceNamespace string
+	dnsDomain                string
 )
 
+func getDNSDomain() string {
```

Comment:
Can we please get a comment explaining how this is operating and why?  It's PROBABLY safe, all things considered, but it's a heuristic.

This has come up in other places, and we really ought to expose it through downward API or something like that.

Reply:
Maybe it is good to have comments, but even this module has more complex functions without comments.
According to the activity on the bug-report which gets fixed here, no one really cared about it. So, I wonder exactly where it has come up?

Reply:
Past mistakes do not mean we shouldn't do better.  The function is sufficiently subtle that having an explanation of what it is supposed to be doing and why is justified.  I'm not asking for a book, just a couple sentences. Less than this comment, probably.

Reply:
Added comment in the code here and in the place of it's call.